services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -4,171 +4,64 @@
-        raise ProjectInvalidRightsError(user_id, project.get("uuid"))
-
-    project_access_rights = deepcopy(project.get("access_rights", {}))
-
-    # compute access rights
-    no_access_rights = {"read": False, "write": False, "delete": False}
-    computed_permissions = project_access_rights.get(
-        str(all_group["gid"]), no_access_rights
-    )
-
-    # get the standard groups
-    for group in standard_groups:
-        standard_project_access = project_access_rights.get(
-            str(group["gid"]), no_access_rights
-        )
-        for k in computed_permissions.keys():
-            computed_permissions[k] = (
-                computed_permissions[k] or standard_project_access[k]
-            )
-
-    # get the primary group access
-    primary_access_right = project_access_rights.get(
-        str(primary_group["gid"]), no_access_rights
-    )
-    for k in computed_permissions.keys():
-        computed_permissions[k] = computed_permissions[k] or primary_access_right[k]
-
-    if any(not computed_permissions[p] for p in needed_permissions):
-        raise ProjectInvalidRightsError(user_id, project.get("uuid"))
-
-
-def _create_project_access_rights(
-    gid: int, access: ProjectAccessRights
-) -> dict[str, dict[str, bool]]:
-    return {f"{gid}": access.value}
-
-
-# TODO: check here how schema to model db works!?
-def _convert_to_db_names(project_document_data: dict) -> dict:
-    converted_args = {}
-    exclude_keys = [
-        "tags",
-        "prjOwner",
-    ]  # No column for tags, prjOwner is a foreign key in db
-    for key, value in project_document_data.items():
-        if not key in exclude_keys:
-            converted_args[camel_to_snake(key)] = value
-    return converted_args
-
-
-def _convert_to_schema_names(
-    project_database_data: Mapping, user_email: str, **kwargs
-) -> dict:
-    converted_args = {}
-    for key, value in project_database_data.items():
-        if key in DB_EXCLUSIVE_COLUMNS:
-            continue
-        converted_value = value
-        if isinstance(value, datetime):
-            converted_value = format_datetime(value)
-        elif key == "prj_owner":
-            # this entry has to be converted to the owner e-mail address
-            converted_value = user_email
-
-        if key in SCHEMA_NON_NULL_KEYS and value is None:
-            converted_value = ""
-
-        converted_args[snake_to_camel(key)] = converted_value
-    converted_args.update(**kwargs)
-    return converted_args
-
-
-def _assemble_array_groups(user_groups: list[RowProxy]) -> str:
-    return (
-        "array[]::text[]"
-        if len(user_groups) == 0
-        else f"""array[{', '.join(f"'{group.gid}'" for group in user_groups)}]"""
-    )
 
 
-# NOTE: https://github.com/ITISFoundation/osparc-simcore/issues/3516
 # pylint: disable=too-many-public-methods
+# NOTE: https://github.com/ITISFoundation/osparc-simcore/issues/3516
 
 
-class ProjectDBAPI:
+class ProjectDBAPI(ProjectDBMixin):
```

Comment:
I am not sure I get the idea behind that mixin and why it is a base class. This basically is a bunch of static method for accessing tables that are not the projects table right?

Reply:
As a personal preference, I would have left them as private functions part of the module where the class lives.

Reply:
changed

Reply:
@GitHK the idea was to move some code away so we could start see at least the interface. 
After doing this exercise, it already revealed how much functionality is duplicated !!!
Next iterations will unify and slowly bring back some of the helpers to this class.